src/app/components/authmodel/AuthModel.js

```diff
@@ -0,0 +1,226 @@
+/* eslint-disable react/prop-types */
+'use client'
+
+import { useState, useEffect, React } from 'react'
+import { signIn, signUp,  confirmSignUp} from 'aws-amplify/auth'
+
+const AuthModel = ({ showModal, setShowModal, authStates, setAuthStates, setIsSignin}) => {
+
+  if (!showModal) return null
+
+  const [username, setUsername] = useState('')
+  const obj = {
+    login: <LoginForm setShowModal={setShowModal} setAuthStates={setAuthStates} username={username} setUsername={setUsername} setIsSignin={setIsSignin}/>, 
+    signup: <RegisterForm setAuthStates={setAuthStates} username={username} setUsername={setUsername} />, 
+    confirm:<ConfirmSignUpForm setAuthStates={setAuthStates} username={username}/>
+}
+
+  return (
+    <div className="w-screen z-30 h-screen fixed inset-0 bg-gray-800 bg-opacity-50 flex items-center justify-center">
+      <div className="bg-white p-6 rounded-lg shadow-lg w-96 relative z-50">
+        <button className="absolute top-2 right-2 text-gray-500 hover:text-gray-700"
+          onClick={() => setShowModal(false)} >X</button>
+        {obj[authStates]}
+
+        <button
+          className="mt-4 text-blue-500 hover:text-blue-700"
+          onClick={() => setAuthStates(authStates==='login'?'signup':'login')}
+        >
+          {authStates==='login' ? 'Switch to Register' : authStates==='signup' ? 'Switch to Login' : ''}
```

Comment:
Please consider hiding this button instead of having an empty button, because it's still possible to client event if it's blank